Messaging.md

```diff
@@ -165,10 +173,69 @@ This type contains signatures of the funding transaction and any necessary infor
 `witness` is the data for a witness element in a witness stack. An empty `witness_stack` is an error,
 as every input must be Segwit. Witness elements should *not* include their length as part of the witness data.
 
+### The `event_descriptor` Type
+
+This type contains information about the *exact* and fully specified outcomes  in an event for which an oracle plans on releasing a signature over.
+
+#### Version 0 `external_event_descriptor`
+
+1. type: 55300 (`external_event_descriptor_v0`)
+2. data:
+   * [`string`:`external_name`]
+
+`external_name` can refer to anything here and it is up to the oracle and user to agree on how to interpret it.
+
+#### Version 0 `enum_event_descriptor`
+
+1. type: 55302 (`enum_event_descriptor_v0`)
+2. data:
+   * [`u16`:`outcome_1_len`]
+   * [`utf8_string`:`outcome_1`]
+   * ...
+   * [`u16`:`outcome_n_len`]
+   * [`utf8_string`:`outcome_n`]
+
+This type of event descriptor is a simple enumeration where the value `n` is the number of outcomes in the event. `n` is not given because it can be found by parsing each outcome's length followed by the actual outcome until you has parsed the same number of bytes as the length field of the TLV.
+
+Each `outcome_i` corresponds to the pre-image of a possible outcome that the oracle could sign.
+
+### The `oracle_event` Type
+
+This type contains information about an event for which an oracle plans on releasing a signature over.
+
+#### Version 0 `oracle_event`
+
+1. type: 55330 (`oracle_event_v0`)
+2. data:
+   * [`x_point`:`oracle_public_key`]
+   * [`x_point`:`oracle_nonce`]
+   * [`u32`:`event_maturity_epoch`]
+   * [`event_descriptor`:`event_descriptor`]
+   * [`utf8_string`:`event_uri`]
+
+`event_maturity_epoch` refers to the earliest time this event (UTC) is expected to be signed, in epoch seconds.
+
+`event_uri` is a name and/or categorization of this event given by the oracle.
+
+### The `oracle_announcement` Type
+
+This type contains information about an announcement of an oracle to attest to an event in the future.
+
+#### Version 0 `oracle_announcement`
+
+1. type: 55332 (`oracle_announcement`)
+2. data:
+   * [`signature`:`annoucement_signature`]
+   * [`oracle_event`:`oracle_event`]
+
+The `annoucement_signature` is a signature of the hash of the serialized `oracle_event` using the `oracle_public_key`. This can be shared with peers that have already verified this oracle's public key.
```

Comment:
> using the `oracle_public_key`

that is valid with respect to `oracle_public_key`.

Might be picky but sounds weird to say you make a signature using a public key.

Reply:
I like this a lot better. I agree it sounded like you signed the pub key but didn't know how to say it better.